pkg/envoy/builders/common/builder.go

```diff
@@ -0,0 +1,30 @@
+package common
+
+type Builder[R any] struct {
+	configurers []Configurer[R]
+}
+
+type Configurer[R any] func(*R) error
+
+func (b *Builder[R]) Configure(configurer Configurer[R]) *Builder[R] {
+	b.configurers = append(b.configurers, configurer)
+	return b
+}
+
+func (b *Builder[R]) ConfigureIf(condition bool, fn func() Configurer[R]) *Builder[R] {
```

Comment:
is there possibility for this condition to be something more complex than a bool? Like function that checks previous configurers output or something like this?